src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Communication\Form\ConnectionSetupSubForms;
+
+use Generated\Shared\Transfer\CompanyBusinessUnitCriteriaFilterTransfer;
+use Generated\Shared\Transfer\CompanyUserCriteriaFilterTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogConnectionSetupTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use Spryker\Zed\Gui\Communication\Form\Type\SelectType;
+use Spryker\Zed\Kernel\Communication\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\PunchoutCatalogsConfig getConfig()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface getRepository()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Communication\PunchoutCatalogsCommunicationFactory getFactory()
+ */
+class PunchoutCatalogConnectionSetupForm extends AbstractType
+{
+    public const OPTION_COMPANY_USERS_ARRAY = 'option-company-users-array';
+
+    protected const FIELD_LABEL_LOGIN_MODE = 'Login Mode';
+    protected const FIELD_LABEL_COMPANY_USER = 'Single User';
+    protected const FIELD_LABEL_COMPANY_BUSINESS_UNIT = 'Default Business Unit';
+
+    protected const DEPENDENT_GROUP_LOGIN_MODE = 'login-mode';
+
+    protected const LOGIN_MODE_SINGLE_USER = 'single_user';
+    protected const LOGIN_MODE_DYNAMIC_USER = 'dynamic_user';
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $this->addLoginModeField($builder)
+            ->addCompanyBusinessUnitField($builder, [])
```

Comment:
Please, do not pass empty options if they are not used there. Remember about YAGNI.